pkg/ignition/network.go

```diff
@@ -91,13 +91,17 @@ DNS={{ $dnsServer }}
 
 {{- define "rules" }}
 {{- if .FIBRules }}
+{{- $type := .Type }}
+{{- $table := .Table }}
 {{- range $index, $rule := .FIBRules }}
-
 [RoutingPolicyRule]
 {{ if $rule.To }}To={{$rule.To}}{{- end }}
 {{ if $rule.From }}From={{$rule.From}}{{- end }}
 {{ if $rule.Priority }}Priority={{$rule.Priority}}{{- end }}
-{{ if $rule.Table }}Table={{$rule.Table}}{{- end }}
+{{ if and (eq $type "vrf") (not $rule.Table) }}Table={{ $table }}
```

Comment:
Why we need `not $rule.Table`

Reply:
If the rule happens to have a table, then we jump into the else branch and apply that instead. Perhaps we should write a test for that.

As an explanation: VRFs have an l3mdev-table built in, and systemd-networkd fails to use that. This is why I patch that behaviour in via template. But if I define a rule with a specific table, then I'd expect that to be used.

Reply:
there's already a test case for that.